src/EFCore.SqlServer/Storage/Internal/SqlServerConnection.cs

```
@@ -163,10 +165,11 @@ protected virtual string ModifyConnectionString(string userProvidedConnectionStr
         if (connectionStringBuilder.ApplicationName is "Core Microsoft SqlClient Data Provider" or "" or null)
         {
             var efVersion = FileVersionInfo.GetVersionInfo(Assembly.GetExecutingAssembly().Location).ProductVersion;
-            var sqlClientVersion = FileVersionInfo.GetVersionInfo(typeof(SqlConnection).Assembly.Location).ProductVersion;
 
-            connectionStringBuilder.ApplicationName =
-                $"EFCore/{efVersion} ({RuntimeInformation.OSDescription} {RuntimeInformation.OSArchitecture}) {RuntimeInformation.FrameworkDescription} SqlClient/{sqlClientVersion}";
+            _defaultApplicationName ??=
+                $"EFCore/{efVersion} ({RuntimeInformation.OSDescription} {RuntimeInformation.OSArchitecture})";
```

Maybe make sure the length stays below 128 chars to avoid runtime errors?

---

I've added a try/catch around the construction of the entire connection string to make sure that rewriting errors don't bubble up (we just silently ignore and use the original connection string) - so that should take care of that too.

The additional versions (.NET, SqlClient) don't seem terribly important here - the EF version already gives us the approximate versions in any case...